diff --git a/modules/system/src/Plugin/ImageToolkit/GDToolkit.php b/modules/system/src/Plugin/ImageToolkit/GDToolkit.php
index c07c9cf3c5..cf77754314 100644
--- a/modules/system/src/Plugin/ImageToolkit/GDToolkit.php
+++ b/modules/system/src/Plugin/ImageToolkit/GDToolkit.php
@@ -463,7 +463,12 @@ class GDToolkit extends ImageToolkitBase {
    *   IMAGETYPE_* constant (e.g. IMAGETYPE_JPEG, IMAGETYPE_PNG, etc.).
    */
   protected static function supportedTypes() {
-    return [IMAGETYPE_PNG, IMAGETYPE_JPEG, IMAGETYPE_GIF];
+    return [
+      IMAGETYPE_PNG,
+      IMAGETYPE_JPEG,
+      IMAGETYPE_GIF,
+      IMAGETYPE_WEBP,
+    ];
   }
 
 }
