diff --git a/src/Plugin/search_api/backend/SearchApiSolrBackend.php b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
index 037f8c3..9f570f0 100644
--- a/src/Plugin/search_api/backend/SearchApiSolrBackend.php
+++ b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
@@ -4784,4 +4784,17 @@ class SearchApiSolrBackend extends BackendPluginBase implements SolrBackendInter
     return (bool) $this->configuration['skip_schema_check'];
   }

+  /**
+   * Implements the magic __sleep() method.
+   *
+   * Prevents the Solr connector from being serialized. There's no need for a
+   * corresponding __wakeup() because of getSolrConnector().
+   * @see getSolrConnector()
+   */
+  public function __sleep() {
+    $properties = array_flip(parent::__sleep());
+    unset($properties['solrConnector']);
+    return array_keys($properties);
+  }
+
 }
